--- a/geant4_app/src/laue_multiring_bfull_demo.cc
+++ b/geant4_app/src/laue_multiring_bfull_demo.cc
@@ -1191,7 +1191,6 @@
         rings_(rings),
         rockingCurves_(rockingCurves) {
     SetBuildTableFlag(false);
-    SetSecondaryParticle(G4Gamma::GammaDefinition());
     SetProcessSubType(9001);
   }
 
@@ -1210,6 +1209,7 @@
   void InitialiseProcess(const G4ParticleDefinition*) override {
     if (isInitialised_) return;
     isInitialised_ = true;
+    SetSecondaryParticle(G4Gamma::GammaDefinition());
     if (!EmModel(1)) SetEmModel(new BFullLaueEmBookkeepingModel(), 1);
     AddEmModel(1, EmModel(1));
   }
